agent/stats/engine.go

```diff
@@ -398,6 +401,13 @@ func (engine *DockerStatsEngine) getContainerMetricsForTask(taskArn string) ([]*
 			engine.doRemoveContainer(container, taskArn)
 			continue
 		}
+
+		if !container.statsQueue.resetThresholdElapsed(queueResetThreshold) &&
```

Comment:
I find this conditional pretty hard to read.  I think it might be easier like this:

```go
if !container.statsQueue.enoughDatapointsInBuffer() &&
	!container.statsQueue.resetThresholdElapsed(queueResetThreshold)
```

That lets it read more logically as "if there are not enough datapoints and the threshold is not elapsed".